tests/e2e/vsan_stretched_cluster.go

```diff
@@ -1965,4 +1966,497 @@ var _ = ginkgo.Describe("[vsan-stretch-vanilla] vsan stretched cluster tests", f
+	       and wait for all replicas to be running
+	   4.  Change replica count of sts1 and sts2 to 3
+	   5.  Bring down secondary site
+	   6.  Verify that the VMs on the secondary site are started up on the other esx servers in the primary site
+	   7.  Verify there were no issue with replica scale up/down and verify pod entry in CNS volumemetadata for the
+	       volumes associated with the PVC used by statefulsets are updated
+	   8.  Change replica count of sts1 to 5 a sts2 to 1 and verify they are successful
+	   9.  Delete statefulsets and its pvcs created in step 2
+	   10. Bring secondary site up and wait for testbed to be back to normal
+	*/
+	ginkgo.It("Statefulset scale up/down while secondary site goes down"+
+		" when csi driver leader is in secondary site", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Creating StorageClass for Statefulset")
+		// decide which test setup is available to run
+		ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+		scParameters = map[string]string{}
+		scParameters["StoragePolicyName"] = storageThickPolicyName
+		storageClassName = "nginx-sc-thick"
+		csiPodOnSite, nodeName := "", ""
+		ignoreLabels := make(map[string]string)
+
+		scSpec := getVSphereStorageClassSpec(storageClassName, scParameters, nil, "", "", false)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		framework.Logf("Ensuring %s leader is in secondary site", csiDriverContainerName)
+		// Fetch master ip present on that site
+		masterIpOnSite, err := getMasterIpOnSite(ctx, client, false)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// Fetch the name of master node on that site from the IP address
+		k8sNodes, err := client.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for _, node := range k8sNodes.Items {
+			addrs := node.Status.Addresses
+			for _, addr := range addrs {
+				if addr.Type == v1.NodeInternalIP && (net.ParseIP(addr.Address)).To4() != nil &&
+					addr.Address == masterIpOnSite {
+					nodeName = node.Name
+					break
+				}
+			}
+		}
+
+		// Get the name pf csi controller pod running on master node on that site
+		csiPods, err := fpod.GetPodsInNamespace(client, csiSystemNamespace, ignoreLabels)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for _, csiPod := range csiPods {
+			if strings.Contains(csiPod.Name, vSphereCSIControllerPodNamePrefix) &&
+				csiPod.Spec.NodeName == nodeName {
+				csiPodOnSite = csiPod.Name
+			}
+		}
+		// Delete csi controller pods on other masters which is not present on that site
+		deleteCsiControllerPodOnOtherMasters(client, csiPodOnSite)
```

Comment:
add a check in this to verify that all other side car leaders are in the required site